### PR TITLE
[feat] Add Torch Sync Batchnorm handle in sharded DDP

### PR DESCRIPTION
**Pull request by SeanNaren**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/5210 by ensuring that we pass in the correct ddp gpu count for the layer! I've added references to where I've taken the code from (cc @blefaudeux)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by blefaudeux — COMMENTED** [review]
Looks great to me, thanks @SeanNaren ! Would you mind adding a unit test ? CI can run on 4 gpus, that should be enough for a dummy syncbn model :) thanks !

**Comment by blefaudeux** [comment]
(Else I can do that in a week from now @SeanNaren, as you prefer !) thanks already !

**Comment by SeanNaren** [comment]
Absolutely leave it to me!

**Comment by SeanNaren** [comment]
Apologies been on and off, I'll try get this done in the next few days @blefaudeux

**Comment by blefaudeux** [comment]
> Apologies been on and off, I'll try get this done in the next few days @blefaudeux

just back myself, feel free to drag me in as need be !

**Comment by SeanNaren** [comment]
Added a small test to ensure that the handles are added and we can successfully do one forward pass. This fails on master but works on this branch! If you'd prefer I can get a parity test between DDP/SDP but that might be a bit overkill, let me know :)

**Review by blefaudeux — APPROVED** [review]
LGTM, thanks Sean, this is a very welcome addition !

**Comment by SeanNaren** [comment]
Thank you man!
